Example-Crypto-REXX/RXCRYPT1.rexx

```diff
@@ -0,0 +1,2453 @@
+       loop in the event HWIHOST(OFF) fails and invokes fatalError,
+       which will goes through this path again
+    */
+    HWIHOST_ON = FALSE
+    hwiHostRC = hwihost("OFF")
+    say 'HWIHOST("OFF") return code is: ('||hwiHostRC||')'
+
+    if hwiHostRC <> 0 then
+      exit fatalErrorAndCleanup('** unable to turn off HWIHOST **')
+  end
+
+return 0
+
+/*******************************************************/
+/* Function:  GetLocalCPCInfo                          */
+/*                                                     */
+/* Retrieve the uri and target name and SE version     */
+/* associated with the LOCAL CPC                       */
+/* Prime CPCuri and CPCtargetName variables            */
+/* Verify SE version is z16 or higher                  */
+/*                                                     */
+/* Return 0 on success, -1 on error                    */
+/*******************************************************/
+GetLocalCPCInfo:
+
+emptyCPCResponse = '{"cpcs":[]}'
+localCPCfound = FALSE
+
+/* First list the cpcs, then retrieve LOCAL
+   CPC which will have
+           "location":"local"
+
+   GET /api/cpcs
+
+   v1: HWILIST(HWI_LIST_LOCAL_CPC)
+*/
+reqUri = '/api/cpcs'
+CPCInfoResponse = GetRequest(reqUri)
+
+emptyCPCArray = INDEX(CPCInfoResponse, emptyCPCResponse)
+if emptyCPCArray > 0 | CPCInfoResponse = '' Then
+  do
+    return fatalError('** failed to get CPC info **')
+  end
+
+/* Parse the response to obtain the uri
+   and target name associated with CPC
+*/
+call JSON_parseJson CPCInfoResponse
+CPCArray = JSON_findValue(0, "cpcs", HWTJ_ARRAY_TYPE)
+if foundJSONValue(CPCArray) = FALSE then
+  do
+    return fatalError('** failed to retrieve CPCs **')
+  end
+
+ /**************************************************************/
+ /* Determine the number of CPCs represented in the array      */
+ /**************************************************************/
+ CPCs = JSON_getArrayDim(CPCArray)
+ if CPCs <= 0 then
+    return fatalError( '** Unable to retrieve number of CPCs **' )
+
+ /********************************************************************/
+ /* Traverse the CPCs array to populate CPCsList.                    */
+ /* We use the REXX (1-based) idiom  but adjust for the differing    */
+ /* (0-based) idiom of the toolkit.                                  */
+ /********************************************************************/
+ say 'Processing information for '||CPCs||' CPC(s)... searching for LOCAL'
+ do i = 1 to CPCs
+    nextEntryHandle = JSON_getArrayEntry(CPCArray,i-1)
+    CPClocation = JSON_findValue(nextEntryHandle,"location",HWTJ_STRING_TYPE)
+
+    if CPClocation = 'local' then
+      do
+        say 'found LOCAL CPC'
+        if localCPCfound = TRUE then
+          do
+            return fatalError('** found two LOCAL CPCs **')
+          end
+        localCPCfound = TRUE
+
+      CPCversion=JSON_findValue(nextEntryHandle,"se-version",HWTJ_STRING_TYPE)
```

Comment:
nit, indent one space please to line it up with the others

Reply:
fixed